pony.g

```diff
@@ -209,7 +209,7 @@ tuple
   ;
 
 lambdacaptures
-  : ('(' | LPAREN_NEW) lambdacapture (',' lambdacapture)* ')'
+  : ('(' | LPAREN_NEW) (lambdacapture | 'this') (',' (lambdacapture | 'this'))* ')'
```

Comment:
It's not clear to me that we should add this to the `pony.g` file, given that it is not valid syntax.

I was under the impression that the `pony.g` grammar diverged from that defined in `ast/parser.c` in those cases where we allow an invalid syntax just so we can print a better error from the `syntax` pass (just like this PR).

Perhaps an easy way to test whether this impression is correct would be to check if there are indeed other divergences in such cases.  If so, I'd suggest leaving this change out of `pony.g`.